libwasmvm/src/handle_vm_panic.rs

```diff
@@ -5,8 +5,14 @@ use std::any::Any;
 /// We want to provide as much debug information as possible
 /// as those cases are not expected to happen during healthy operations.
 pub fn handle_vm_panic(what: &str, err: Box<dyn Any + Send + 'static>) {
+    let err = match (err.downcast_ref::<&str>(), err.downcast_ref::<String>()) {
+        (Some(str), ..) => *str,
+        (.., Some(str)) => str,
+        (None, None) => "[unusable panic payload]",
+    };
```

Comment:
What type is `err` here? `&String`?

Reply:
It's an `&str`. The lowest common denominator between `&str` and `String`

Reply:
But looking at https://doc.rust-lang.org/std/any/trait.Any.html#method.downcast_ref, isn't `err.downcast_ref::<String>()` returning `&T` which is `&String`?

Reply:
I used a `match` here since the code is just more compact and basically just as readable as an `if` chain. Just that both branches are executed, but it's fine since a downcast is basically:

```rust
if type_id_actual == type_id_potentially {
  Some(pointer_cast(value))
} else {
  None
}
```

Very little work

Reply:
> But looking at https://doc.rust-lang.org/std/any/trait.Any.html#method.downcast_ref, isn't err.downcast_ref::<String>() returning &T which is &String?

Oh, sorry, yeah. That returns `&String` but type coersion will turn it into an `&str`. Yeah, the most common types in a panic payload are `&str` and `String`. So that covers pretty much all bases

Reply:
Yeah I like the idea, especially for testing just two options

Reply:
Ah great, I was not aware Rust does this so implicitly